pkg/heartbeat/util.go

```diff
@@ -35,21 +38,67 @@ func InjectHeartbeatInfoHeaders(req *http.Request, heartbeatInfo *types.Heartbea
 	}
 }
 
-func canReport(license *kotsv1beta1.License) (bool, error) {
-	if helm.IsHelmManaged() {
-		sdkHelmRevision := helm.GetReleaseRevision()
+func canReport(clientset kubernetes.Interface, namespace string, license *kotsv1beta1.License) (bool, error) {
+	deployment, err := clientset.AppsV1().Deployments(namespace).Get(context.TODO(), "replicated-sdk", metav1.GetOptions{})
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get replicated-sdk deployment")
+	}
 
-		helmRelease, err := helm.GetRelease(helm.GetReleaseName())
-		if err != nil {
-			return false, errors.Wrap(err, "failed to get release")
+	podSelector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get pod selector from deployment")
+	}
+
+	podList, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: podSelector.String(),
+	})
+	if err != nil {
+		return false, errors.Wrap(err, "failed to list pods")
+	}
+
+	var podRevision int
+	for _, pod := range podList.Items {
+		if pod.Name != os.Getenv("REPLICATED_SDK_POD_NAME") {
```

Comment:
Can this pod ever not be on the list?  If so, `podRevision` will be uninitialized.  Does that break `podRevision != deploymentRevision`?

Reply:
True. This is a remnant of an alternative logic I had before this. Thanks!